src/utils/snake.ts

```diff
@@ -0,0 +1,32 @@
+import { SNAKE } from '../constants';
+
+import type { IObjectBody } from '../types/canvas';
+
+export const isSnakeCollided = (
+  SnakeBodyPosition: IObjectBody[],
+  SnakeHeadPosition: IObjectBody,
+): boolean => {
+  for (let i = 0; i < SnakeBodyPosition.length; i++) {
+    const isCurrentCollided =
+      SnakeBodyPosition[i].x === SnakeHeadPosition.x &&
+      SnakeBodyPosition[i].y === SnakeHeadPosition.y &&
+      i !== 0;
+    if (isCurrentCollided) return true;
+  }
```

Comment:
```suggestion
    SnakeBodyPosition.forEach(({ x, y }, index) => {
    const isCurrentCollided = 
		x === SnakeHeadPosition.x && 
		y === SnakeHeadPosition.y && 
		index !== 0;

    if (isCurrentCollided) return true;
  });
```

위 for문을 좀 더 js 스럽게 forEach로 사용해보면 어때??

Reply:
이게 for문이랑 forEach문이 다른 점이
`for`문은 어떤 조건이 만족될 때 바로 return 할 수 있는 return을 제공하는데,
`forEach`안에서는 return문을 넣어도 바로 `forEach`가 종료되지 않아.
그리고 무조건 true를 반환하게 되어있어서 원하는대로 동작을 안할거야. 그래서 for문을 사용했으!

Reply:
> 이게 for문이랑 forEach문이 다른 점이 `for`문은 어떤 조건이 만족될 때 바로 return 할 수 있는 return을 제공하는데, `forEach`안에서는 return문을 넣어도 바로 `forEach`가 종료되지 않아. 그리고 무조건 true를 반환하게 되어있어서 원하는대로 동작을 안할거야. 그래서 for문을 사용했으!

오오 그런차이가 있었구나!! 하나또 배워갑니다 😄